### DISCUSSION TITLE
Storing arbitrary bytes in Redis

### DISCUSSION THREAD

**Discussion by cinderrrr** [redis/rueidis#173] [Q&A] [answered]
Hi there-- is there a way to store arbitrarily encoded byte arrays with Get/Set? Or do I need to manage the encoding/decoding on my end?

**Comment by rueian** [comment, marked as answer]
Redis string is binary safe. So yes, you can store arbitrary Go byte slice as Go string without special encoding.

And `rueidis.BinaryString()` can help you do the conversion without memory copy.

**Reply by cinderrrr** [reply]
Is there a clean way to do this with the response of an mget? It looks like the documentation says to use RedisMessage.AsReader(), but does this work for []RedisMessages to get  back to [][]byte?

**Reply by rueian** [reply]
For MGET, I would recommend using AsStrSlice() to get the response as []string first. And then view each string as []byte while iterating.

**Reply by 0x079** [reply]
I think casting string to []byte will cause allocation?

**Reply by rueian** [reply]
It depends. But indeed, casting with full allocation is the most easiest way.

**Reply by cinderrrr** [reply]
I ended up having to use `AsArray(),` because I needed to know whether the call was a hit, and we sometimes have empty string values.
